types/terser-webpack-plugin/index.d.ts

```diff
@@ -23,32 +20,104 @@ declare namespace TerserPlugin {
+         * @default undefined
+         */
         exclude?: string | RegExp | Array<string | RegExp>;
-        chunkFilter?: (chunk: webpack.compilation.Chunk) => boolean;
+
+        /**
+         * ⚠ Ignored in webpack 5! Please use {@link webpack.js.org/configuration/other-options/#cache.}
+         * Enable/disable file caching.
+         * Default path to cache directory: `node_modules/.cache/terser-webpack-plugin`.
+         * @default true
+         */
         cache?: boolean | string;
+
+        /**
+         * ⚠ Ignored in webpack 5! Please use {@link webpack.js.org/configuration/other-options/#cache}.
+         * Allows you to override default cache keys.
+         */
         cacheKeys?: (defaultCacheKeys: any, file: any) => object;
+
+        /**
+         * Enable/disable multi-process parallel running.
+         * Use multi-process parallel running to improve the build speed. Default number of concurrent runs: os.cpus().length - 1.
+         * @default true
+         */
         parallel?: boolean | number;
+
+        /**
+         * Use source maps to map error message locations to modules (this slows down the compilation).
+         * If you use your own minify function please read the minify section for handling source maps correctly.
+         * @default false
+         */
         sourceMap?: boolean;
+
+        /**
+         * Allows you to override default minify function.
+         * By default plugin uses terser package. Useful for using and testing unpublished versions or forks
+         * @default undefined
+         */
         minify?: (file: any, sourceMap: any) => MinifyResult;
+
+        /**
+         * Terser minify {@link https://github.com/terser/terser#minify-options|options}.
+         */
         terserOptions?: MinifyOptions;
-        extractComments?: boolean
-        | string
-        | RegExp
-        | ExtractCommentFn
-        | ExtractCommentOptions;
-        warningsFilter?: (warning: string, source: string | undefined, file: string) => boolean;
+
+        /**
+         * Whether comments shall be extracted to a separate file, (see details).
+         * By default extract only comments using /^\**!|@preserve|@license|@cc_on/i regexp condition and remove remaining comments.
+         * If the original file is named foo.js, then the comments will be stored to foo.js.LICENSE.txt.
+         * The terserOptions.output.comments option specifies whether the comment will be preserved,
+         * i.e. it is possible to preserve some comments (e.g. annotations) while extracting others or even preserving comments that have been extracted
+         * @default true
+         */
+        extractComments?: boolean | string | RegExp | ExtractCommentFn | ExtractCommentOptions;
+
+        /**
+         * Allow to filter terser warnings.
+         * @default () => true
+         */
+        warningsFilter?: (file: string, warning: string, source: string) => true | false | null | undefined;
```

Comment:
For brevity: Please use `boolean` instead of `true | false`.
The order of parameters is wrong! [documentation](https://webpack.js.org/plugins/terser-webpack-plugin/#warningsfilter)
Also:
> The source argument will contain undefined if you don't use source maps.

Reply:
amended with d058505e3d

this order of params, I was influenced by their readme:
> change arguments order for the warningFilter option from Function<(warning, file, source) -> Boolean> to Function<(file, warning, source) -> Boolean>

while their source code has order of params applied now:
https://github.com/webpack-contrib/terser-webpack-plugin/blob/master/src/index.js#L159